runtime/src/main/java/org/corfudb/runtime/collections/TableSchema.java

```diff
@@ -0,0 +1,31 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Message;
+import lombok.EqualsAndHashCode;
+
+/**
+ * Definition of the fully qualified Corfu Store Table initialized with default protobuf instances.
+ * This class is necessary for StreamListener's deserialization of the stream entries.
+ * It also allows a way to apply a filter and only specify the tables of interest to streaming.
+ *
+ * @param <K> - Default instance type of the protobuf generated key schema
+ * @param <V> - Default instance type of the protobuf generated payload schema
+ * @param <M> - Default instance type of the protobuf generated metadata schema
+ *
+ * created by hisundar on 2019-10-29
+ */
+
+@EqualsAndHashCode
+public class TableSchema<K extends Message, V extends Message, M extends Message> {
+    private final String tableName;
+    private final Class<K> keyClass;
+    private final Class<V> payloadClass;
+    private final Class<M> metadataClass;
+
+    public TableSchema(String tableName, Class<K> keyClass, Class<V> payloadClass, Class<M> metadataClass) {
```

Comment:
Can you add `@Nonnull` to the parameters except metadataClass which can be `@Nullable`?

Reply:
done.